### PR TITLE
Fix UnicodeDecodeError in file-viewer and possibly elsewhere.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This essentially reverts #1100 or the behavior introduced by it. I'll update this PR with a more detailed write-up about what and why this changed.

Let's first see what travis says.

### CODE REVIEW

File: src/olympia/files/tests/test_views.py [hunk 1]
```diff
@@ -381,6 +383,14 @@ def test_unicode(self):
...
         assert res.status_code == 200
 
+    @override_settings(TMP_PATH=unicode(settings.TMP_PATH))
+    def test_unicode_fails_with_wrong_configured_basepath(self):
+        file_viewer = FileViewer(self.file)
+        file_viewer.src = unicode_filenames
+
+        with pytest.raises(UnicodeDecodeError):
+            file_viewer.extract()
+
```
Comment (EnTeQuAk) [inline src/olympia/files/tests/test_views.py]: https://github.com/mozilla/addons-server/pull/3880/files#diff-9cd11e480f767233e6e5b2c046cddddcR380 is the original test that tests unicode filenames already so both scenarios are covered.

File: src/olympia/addons/cron.py [hunk 2]
```diff
@@ -226,9 +227,9 @@ def unhide_disabled_files():
     files = set(File.objects.filter(q | Q(status=amo.STATUS_DISABLED))
                 .values_list('version__addon', 'filename'))
     for filepath in walkfiles(settings.GUARDED_ADDONS_PATH):
-        addon, filename = filepath.split('/')[-2:]
+        addon, filename = force_text(filepath).split('/')[-2:]
         if tuple([int(addon), filename]) not in files:
-            log.warning(u'File that should not be guarded: %s.', filepath)
+            log.warning('File that should not be guarded: %s.', filepath)
```
Comment (diox) [inline src/olympia/addons/cron.py]: I'd prefer to `force_text` on `filepath` at the beginning of the loop and keep the `u` prefix there and below.

**Comment by EnTeQuAk** [comment]
Good idea, I'll add some checks and another test that tests that all relevant settings are byte-encoded rather than unicode.

**Comment by EnTeQuAk** [comment]
@diox added tests, monitors and lots more docs to the actual test. Should be ready for another check.

File: src/olympia/amo/tests/test_settings.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from django.conf import settings
+
+
+class TestSettingsValidity(object):
```
Comment (diox) [inline src/olympia/amo/tests/test_settings.py]: Do you really need a class ?
